articles/quickstart/spa/auth0-react/01.md

```diff
@@ -0,0 +1,297 @@
+
+With React Router, your goal is to use the `history` object within the `onRedirectCallback` method to navigate to that private route programmatically. React Router exposes the `useHistory()` custom React Hook to make it easy to access the `history` object in any component that is a descendant of the `BrowserRouter` component.
+
+As such, you can create an `Auth0ProviderWithHistory` function component that renders `Auth0Provider`. This wrapper component then calls `useHistory()` within its body to make the `history` object available within the scope of `Auth0Provider` and its  `onRedirectCallback` event handler.
+
+**Root component:**
+
+```javascript
+ReactDOM.render(
+  <BrowserRouter>
+    <Auth0ProviderWithHistory>
+      <App />
+    </Auth0ProviderWithHistory>
+  </BrowserRouter>,
+  document.getElementById("root")
+);
+```
+
+**`Auth0ProviderWithHistory` component:**
+
+```javascript
+const Auth0ProviderWithHistory = ({ children }) => {
+  const history = useHistory();
+
+  const onRedirectCallback = (appState) => {
+    history.push(appState?.returnTo || window.location.pathname);
+  };
+
+  return (
+    <Auth0Provider
+      domain="${account.namespace}"
+      client_id="${account.clientId}"
+      redirect_uri={window.location.origin}
+      onRedirectCallback={onRedirectCallback}
+    >
+      {children}
+    </Auth0Provider>
+  );
+};
+```
+
+The implementation of the authentication route guard may look like this `PrivateRoute` component:
+
+```javascript
+const PrivateRoute = ({ component, ...args }) => (
+  <Route component={withLoginRequired(component)} {...args} />
+);
+```
+
+You can use the `PrivateRoute` component in the same way that you use the `Route` component from React Router:
+
+```javascript
+function App() {
+  const { isLoading } = useAuth0();
+
+  return (
+    <div className="App">
+      {isLoading ? (
+        <h2>Loading...</h2>
+      ) : (
+        <>
+          <NavBar />
+          <Switch>
+            <Route path="/" exact component={Home} />
+            <PrivateRoute path="/profile" component={Profile} />
+          </Switch>
+        </>
+      )}
+    </div>
+  );
+}
+```
+
+⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️
+
+**TODO: Final URL to sample.**
+ 
+⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️ ⚠️
+
+Take a look at this sample that puts this technique together while also implementing a navigation bar: "Using React Router with the Auth0 React SDK" [sample required]
+
+## Call an API
+
+Your React application may need to access protected data from an API that you are also [securing with Auth0](https://auth0.com/docs/microsites/protect-api/protect-api). For that use case, you'll need to retrieve an access token in the format of a [JSON Web Token (JWT)](https://jwt.io/) to authorize any calls you make to that secured API. 
```

Comment:
This section doesn't say anything about scopes.

Reply:
Sandrino, from your perspective, what should we cover about scopes in the QuickStart? One of the goals is to keep it short and quick as to demonstrate quickly the minimum integration of Auth0 with XYZ (in this case, React). We've felt that more nuanced and detailed concepts should be delegated to documentation and their practice to blog posts.

Reply:
Ok let's add a link to some page that talks about scopes then. However, the SDK works in a specific way when it comes to scopes, we should also have a link to that.

Reply:
Moved to CF.